statsmodels/regression/linear_model.py

```diff
@@ -330,6 +332,40 @@ def fit(self, method="pinv", cov_type='nonrobust', cov_kwds=None,
             # used in ANOVA
             self.effects = effects = np.dot(Q.T, self.wendog)
             beta = np.linalg.solve(R, effects)
+        elif method == "qr-pivot":
+            Q, R, P = linalg.qr(self.wexog, mode='economic', pivoting=True)
+            self.exog_Q, self.exog_R, self.exog_P = Q, R, P
+
+            # Cache singular values from R.
+            self.wexog_singular_values = np.linalg.svd(R, 0, 0)
+            if 'tol' in kwargs:
+                tol = kwargs['tol']
+            else:
+                tol = 1e-7
+
+            singular_preds = abs(np.diagonal(R)) < tol
+            n_singular = sum(singular_preds)
+            n_preds = len(singular_preds)
+            self.rank = n_preds - n_singular
+            sortP = np.argsort(P)
+
+            if n_singular > 0:
+                rng = range(self.rank)
+                Qq = Q[:, rng]
+                Rr = R[rng, :][:, rng]
+                effects = np.dot(Qq.T, self.wendog)
+                beta = linalg.solve_triangular(Rr, effects)
+                beta = np.concatenate((beta, np.repeat(0.0, n_singular)))[sortP]
```

Comment:
I believe there is no solution that will be able to be a  “perfect” choice of dealing with redundant variables. The options are: 
        - setting the value of the redundant variables to NA
	- removing these variables
	- setting the variable to 0
I’ve picked the last choice ‘0’ is the most consistent choice for the following reasons:
- It is consistent with regularization method that “trim” or “remove” variables from the model (LASSO for example), although it is not a regularization algorithm
- It is much more practical compared of setting NA. A user already using statsmodels regression will not expect to get coefficient with NA values. The efforts that current users will need to convert any existing code to the ‘qr-pivot’ with NA coefficient values is almost impossible.
- The same practical issues will raise for the solution of removing the variables. 
- And conceptually these variable contribute nothing “0” by adding them to the model because thay have full multi-collinearity 

Considering all options (and there is not optimal solution for this issue), it would make more sense for me to keep the variables and set the coefficient values to 0.